saleor/graphql/discount/mutations.py

```diff
@@ -382,4 +396,117 @@ def perform_mutation(cls, _root, info, **data):
             info, data.get("id"), only_type=Sale, field="sale_id"
         )
         cls.remove_catalogues_from_node(sale, data.get("input"))
-        return SaleRemoveCatalogues(sale=sale)
+        return SaleRemoveCatalogues(sale=ChannelContext(node=sale, channel_slug=None))
+
+
+class SaleChannelListingAddInput(graphene.InputObjectType):
+    channel_id = graphene.ID(required=True, description="ID of a channel.")
+    discount_value = PositiveDecimal(
+        required=True, description="The value of the discount."
+    )
+
+
+class SaleChannelListingInput(graphene.InputObjectType):
+    add_channels = graphene.List(
+        graphene.NonNull(SaleChannelListingAddInput),
+        description="List of channels to which the sale should be assigned.",
+        required=False,
+    )
+    remove_channels = graphene.List(
+        graphene.NonNull(graphene.ID),
+        description=("List of channels from which the sale should be unassigned."),
+        required=False,
+    )
+
+
+class SaleChannelListingUpdate(BaseChannelListingMutation):
+    sale = graphene.Field(Sale, description="An updated sale instance.")
+
+    class Arguments:
+        id = graphene.ID(required=True, description="ID of a sale to update.")
+        input = SaleChannelListingInput(
+            required=True,
+            description="Fields required to update sale channel listings.",
+        )
+
+    class Meta:
+        description = "Manage sale's availability in channels."
+        permissions = (DiscountPermissions.MANAGE_DISCOUNTS,)
+        error_type_class = DiscountError
+        error_type_field = "discount_errors"
+
+    @classmethod
+    def add_channels(cls, sale: "SaleModel", add_channels: List[Dict]):
+        for add_channel in add_channels:
+            defaults = {}
+            discount_value = add_channel.get("discount_value")
+            channel = add_channel["channel"]
+            currency_code = channel.currency_code
+            if discount_value:
+                defaults["discount_value"] = discount_value
+            SaleChannelListing.objects.update_or_create(
+                sale=sale, channel=channel, currency=currency_code, defaults=defaults,
+            )
+
+    @classmethod
+    def clean_prices(cls, cleaned_channels, sale_type, errors, error_code):
+        if sale_type != DiscountValueType.FIXED:
```

Comment:
Can we add a test for `Percentage` discount we currency with less than 3 decimal precision and value for example 1.005?

Reply:
after a conversation with @fowczarek, we will skip this test for now.